bundles/org.openhab.binding.unifiedremote/src/main/java/org/openhab/binding/unifiedremote/internal/UnifiedRemoteDiscoveryService.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.unifiedremote.internal;
+
+import static org.openhab.binding.unifiedremote.internal.UnifiedRemoteBindingConstants.*;
+
+import java.io.IOException;
+import java.net.*;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link UnifiedRemoteDiscoveryService} discover Unified Remote Server Instances in the network.
+ *
+ * @author Miguel Alvarez - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.unifiedremote")
+@NonNullByDefault
+public class UnifiedRemoteDiscoveryService extends AbstractDiscoveryService {
+
+    private Logger logger = LoggerFactory.getLogger(UnifiedRemoteDiscoveryService.class);
+    static final int TIMEOUT_MS = 20000;
+    private static final long DISCOVERY_RESULT_TTL_SEC = TimeUnit.MINUTES.toSeconds(5);
+
+    public UnifiedRemoteDiscoveryService() {
+        super(SUPPORTED_THING_TYPES, TIMEOUT_MS, false);
+    }
+
+    @Override
+    protected void startScan() {
+        UnifiedRemoteUdpDiscovery client = new UnifiedRemoteUdpDiscovery();
+        client.call(this::addNewServer);
+    }
+
+    private void addNewServer(UnifiedRemoteUdpDiscovery.ServerInfo serverInfo) {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(PARAMETER_MAC_ADDRESS, serverInfo.macAddress);
+        properties.put(PARAMETER_HOSTNAME, serverInfo.host);
+        properties.put(PARAMETER_TCP_PORT, serverInfo.tcpPort);
+        properties.put(PARAMETER_UDP_PORT, serverInfo.udpPort);
+        thingDiscovered(
+                DiscoveryResultBuilder.create(new ThingUID(THING_TYPE_UNIFIED_REMOTE_SERVER, serverInfo.macAddress))
+                        .withTTL(DISCOVERY_RESULT_TTL_SEC).withRepresentationProperty(PARAMETER_MAC_ADDRESS)
+                        .withProperties(properties).withLabel(serverInfo.name).build());
+    }
+
+    private class UnifiedRemoteUdpDiscovery {
```

Comment:
Is this nested class even needed? It doesn't have any fields. Can you just move all this logic into the UnifiedRemoteDiscoveryService class instead?